docs/api.md

```diff
@@ -2351,6 +2375,34 @@ Only one trace can be active at a time per browser.
 #### tracing.stop()
 - returns: <[Promise]<[Buffer]>> Promise which resolves to buffer with trace data.
 
+### class: FileChooser
+
+[FileChooser] objects are dispatched by page via the ['page.interceptFileChooser'](#page-interceptfilechooser) method.
+
+File choosers let you react to page requesting for file selection.
+
+An example of using [FileChooser]:
+
+```js
+const [fileChooser] = await Promise.all([
+  page.interceptFileChooser(),
+  page.click('#upload-file-button'), // some button that triggers file selection
+]);
+await fileChooser.accept(['/tmp/myfile.pdf']);
+```
+
+#### fileChooser.accept(filePaths)
+- `filePaths` <[Array]<[string]>> Accept file picker request with given paths. If some of the  `filePaths` are relative paths, then they are resolved relative to the [current working directory](https://nodejs.org/api/process.html#process_process_cwd).
+- returns: <[Promise]>
+
+#### fileChooser.cancel()
+- returns: <[Promise]>
+
+Cancel file chooser requests. As a result, a follow-up file choosers might be shown.
```

Comment:
What does this mean? When should I call this?

Reply:
whenever you want cancel file chooser dialog. How do you wish this worded?

Reply:
Closes the file chooser without selecting any files.
Somewhere:
> Note: All file choosers must be accepted or canceled. Not doing so will prevent subsequent file choosers from appearing.

Whats the difference between accept([]) and cancel()? Maybe we can merge them into one method, like `fileChooser.respond`?

Reply:
> Whats the difference between accept([]) and cancel()? Maybe we can merge them into one method, like fileChooser.respond?

The `fileChooser.cancel()` is a user cancelling file chooser, which is a vendor-defined behavior. In Chrome, this will also reset selected filed; it will not however do so in Firefox.

The `fileChooser.accept([])` is a vendor-agnostic way to reset input value.